lib/libft/src/ft_strsplit.c

```diff
@@ -0,0 +1,84 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_strsplit.c                                      :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: albarret <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/11/12 16:17:20 by albarret          #+#    #+#             */
+/*   Updated: 2018/11/12 22:51:14 by albarret         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+#include <stdlib.h>
+
+static int		ft_wordcount(char const *s, char c)
+{
+	int i;
+	int count;
+
+	i = 0;
+	count = 0;
+	while (s[i])
+	{
+		while (s[i] == c)
+			i++;
+		if (s[i] != c && s[i])
+			count++;
+		while (s[i] != c && s[i])
+			i++;
+	}
+	return (count);
+}
+
+static int		ft_wordsize(char const *str, char c)
+{
+	int	i;
+	int	len;
+
+	i = 0;
+	len = 0;
+	while (str[i] == c)
+		i++;
+	while (str[i] != c && str[i] != '\0')
+	{
+		i++;
+		len++;
+	}
+	return (len);
+}
+
+static char		**ft_split(char const *s, char c)
+{
+	char **mem;
+	int i;
+	int j;
+	int k;
+
+	i = -1;
+	j = 0;
+	if (!s || !c || !(mem = (char**)malloc(sizeof(char*) * (ft_wordcount(s, c) + 1))))
+		return (NULL);
+	while (ft_wordcount(s, c) > ++i)
+	{
+		k = 0;
+		if (!(mem[i] = (char*)malloc(sizeof(char) * (ft_wordsize(&s[j], c) + 1))))
+			return (NULL);
```

Comment:
In the event of a failed memory allocation, previous allocations made by this function are leaked.

Reply:
Resolving this since it's obsolete.